runtime/lua/vim/diagnostic.lua

```diff
@@ -1603,6 +1616,271 @@ M.handlers.virtual_text = {
   end,
 }
 
+local virtual_lines_augroup =
+  api.nvim_create_augroup('vim_lsp_diagnostic_virt_lines', { clear = true })
```

Comment:
Rather than defining a single augroup at the top-level scope, would it make sense to make the augroup ID a field of `ns.user_data` in the same way that we do for the extmark's namespace?

Reply:
I guess we could. This is what `lsp_lines` does. @WhyNotHugo is there a reason why [this `LspLines` augroup](https://git.sr.ht/~whynothugo/lsp_lines.nvim/tree/a92c755f182b89ea91bd8a6a2227208026f27b4d/item/lua/lsp_lines/init.lua#L32) is created at the top level?

Reply:
No reason, this was simply the easiest way to do it